pkg/injector/envoy_config_health_probes.go

```diff
@@ -63,100 +86,229 @@ func getProbeCluster(clusterName string, port int32) map[string]interface{} {
 	}
 }
 
-func getLivenessListener(originalProbe *healthProbe) map[string]interface{} {
+func getLivenessListener(originalProbe *healthProbe) *xds_listener.Listener {
 	if originalProbe == nil {
 		return nil
 	}
 	return getProbeListener(livenessListener, livenessCluster, livenessProbePath, livenessProbePort, originalProbe)
 }
 
-func getReadinessListener(originalProbe *healthProbe) map[string]interface{} {
+func getReadinessListener(originalProbe *healthProbe) *xds_listener.Listener {
 	if originalProbe == nil {
 		return nil
 	}
 	return getProbeListener(readinessListener, readinessCluster, readinessProbePath, readinessProbePort, originalProbe)
 }
 
-func getStartupListener(originalProbe *healthProbe) map[string]interface{} {
+func getStartupListener(originalProbe *healthProbe) *xds_listener.Listener {
 	if originalProbe == nil {
 		return nil
 	}
 	return getProbeListener(startupListener, startupCluster, startupProbePath, startupProbePort, originalProbe)
 }
 
-func getProbeListener(listenerName, clusterName, newPath string, port int32, originalProbe *healthProbe) map[string]interface{} {
-	filters := []map[string]interface{}{
-		{
-			"name": "envoy.filters.network.http_connection_manager",
-			"typed_config": map[string]interface{}{
-				"@type":       "type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager",
-				"stat_prefix": "health_probes_http",
-				"access_log":  getHTTPAccessLog(),
-				"codec_type":  "AUTO",
-				"route_config": map[string]interface{}{
-					"name":          "local_route",
-					"virtual_hosts": getVirtualHosts(newPath, clusterName, originalProbe.path),
+func getProbeListener(listenerName, clusterName, newPath string, port int32, originalProbe *healthProbe) *xds_listener.Listener {
+	var filterChain *xds_listener.FilterChain
+	if originalProbe.isHTTP {
+		httpConnectionManager := &xds_http_connection_manager.HttpConnectionManager{
+			CodecType:  xds_http_connection_manager.HttpConnectionManager_AUTO,
+			StatPrefix: "health_probes_http",
+			AccessLog:  GetAccessLog(),
+			RouteSpecifier: &xds_http_connection_manager.HttpConnectionManager_RouteConfig{
+				RouteConfig: &xds_route.RouteConfiguration{
+					Name:         "local_route",
+					VirtualHosts: getVirtualHosts(newPath, clusterName, originalProbe.path),
+				},
+			},
+			HttpFilters: []*xds_http_connection_manager.HttpFilter{
+				{
+					Name: "envoy.filters.http.router",
 				},
-				"http_filters": []map[string]interface{}{
-					{
-						"name": "envoy.filters.http.router",
+			},
+		}
+		pbHTTPConnectionManager, err := ptypes.MarshalAny(httpConnectionManager)
+		if err != nil {
+			log.Error().Err(err).Msgf("Error marshaling HttpConnectionManager struct into an anypb.Any message")
+			return nil
```

Comment:
nit: have this function return an error else health probes functionality could be broken since an `append()` with `nil` value is valid.

Reply:
Should the getters also return an error to be handled in `getStaticResources` in `envoy_config.go`?

Reply:
I would suggest returning an error from functions that can generate errors. In some cases such as this one where `MarshalAny` won't result in an error (because of correctness in the http connection manager struct), it is more about correctness.